google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -3420,23 +3407,37 @@ public void testUploadWithEncryption() throws Exception {
   }
 
   @Test
-  public void testGetNotification() {
-    Notification actualNotification =
-        storage.getNotification(BUCKET, notification.getGeneratedId());
-    assertEquals(CUSTOM_ATTRIBUTES, actualNotification.getCustomAttributes());
-    assertEquals(PAYLOAD_FORMAT.name(), actualNotification.getPayloadFormat().name());
-    assertTrue(actualNotification.getTopic().contains(TOPIC));
-  }
-
-  @Test
-  public void testListNotification() {
-    List<Notification> notifications = storage.listNotifications(BUCKET);
-    for (Notification actualNotification : notifications) {
-      if (actualNotification.getGeneratedId().equals(notification.getGeneratedId())) {
-        assertEquals(CUSTOM_ATTRIBUTES, actualNotification.getCustomAttributes());
-        assertEquals(PAYLOAD_FORMAT.name(), actualNotification.getPayloadFormat().name());
-        assertTrue(actualNotification.getTopic().contains(TOPIC));
-      }
+  public void testNotification() throws InterruptedException, ExecutionException {
+    String bucketName = RemoteStorageHelper.generateBucketName();
+    Bucket bucket = storage.create(BucketInfo.newBuilder(bucketName).setLocation("us").build());
+    Notification bucketNotification =
+        Notification.newBuilder(TOPIC)
+            .setCustomAttributes(CUSTOM_ATTRIBUTES)
+            .setPayloadFormat(PAYLOAD_FORMAT)
+            .build();
+    try {
+      Notification notification = storage.addNotification(bucketName, bucketNotification);
+      assertThat(notification.getGeneratedId()).isNotNull();
+      assertThat(CUSTOM_ATTRIBUTES).isEqualTo(notification.getCustomAttributes());
+      assertThat(PAYLOAD_FORMAT.name()).isEqualTo(notification.getPayloadFormat().name());
+      assertThat(notification.getTopic().contains(TOPIC)).isTrue();
+
+      // Gets the notification with the specified id.
+      Notification actualNotification =
+          storage.getNotification(bucketName, notification.getGeneratedId());
+      assertEquals(notification, actualNotification);
+
+      // Retrieves the list of notifications associated with the bucket.
+      List<Notification> notifications = storage.listNotifications(bucketName);
+      assertEquals(1, notifications.size());
+
+      // Deletes the notification with the specified id.
+      boolean isDeleted = storage.deleteNotification(bucketName, notification.getGeneratedId());
+      assertThat(isDeleted).isTrue();
+      assertThat(storage.getNotification(bucketName, notification.getGeneratedId())).isNull();
+      assertThat(storage.listNotifications(bucketName)).isEmpty();
+    } finally {
```

Comment:
add check that storage.getNotification(bucketName, notification.getGeneratedId()) returns null.

Reply:
its already there `assertThat(storage.getNotification(bucketName, notification.getGeneratedId())).isNull();`

Reply:
yes, I missed that